consumer/kinesis.go

```diff
@@ -320,6 +323,38 @@ func (cons *Kinesis) connect() error {
 		Log.Debug.Printf("Starting kinesis consumer for %s:%s", cons.stream, shardID)
 		go cons.processShard(shardID)
 	}
+
+	if cons.shardTime {
+		return nil
+	}
+
+	for cons.running {
```

Comment:
Bad idea, as line 374 will never be called.
Suggestion: Use time.AfterFunc for this